src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseAcquirer.java

```diff
@@ -42,10 +44,29 @@ public void ifAcquiredOrElse(
             var leaseClient = leaseClientProvider.get(blobClient);
             var leaseId = leaseClient.acquireLease(LEASE_DURATION_IN_SECONDS);
 
-            onSuccess.accept(leaseId);
+            boolean isReady = false;
 
-            if (releaseLease) {
-                release(leaseClient, blobClient);
+            try {
+                isReady = blobMetaDataHandler.isBlobReadyToUse(blobClient, leaseId);
+            } catch (Exception ex) {
+                logger.warn(
+                    "Could not check meta data for lease expiration on file {} in container {}",
+                    blobClient.getBlobName(),
+                    blobClient.getContainerName()
+                );
+            } finally {
```

Comment:
We do not need this `finally` clause, it is just overcomplicates the same logic as:
`if (isReady) {
                onSuccess.accept(leaseId);
                if (releaseLease) {
                    clearMetadataAndReleaseLease(leaseClient, blobClient, leaseId);
                }
} else {
                    release(leaseClient, blobClient);
                    //it means lease did not acquired let the failure function decide
                    onFailure.accept(LEASE_ALREADY_PRESENT);
                }
}
`

Reply:
we need to catch exception

Reply:
yes, catch exception and after `catch` clause - this
```
if {
...
} else {
...
}
```